### PR TITLE
Support CREATE, DELETE, MODIFY rules

### LINKED ISSUE #28: Support CREATE, DELETE, MODIFY
**Description of issue or feature request**:
The current set of rules supported by this implementation doesn't include `CREATE`, `DELETE`, and `MODIFY`. 

**Current behavior**: These rules aren't parsed accordingly (which leads to failure).

**Expected behavior**: These rules should be processed according to the spefcification.

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
Closes #28 
Includes changes PRed in #34 (either review and merge #34 first, or only review here)

**Description of pull request**:
Add support for missing rule types CREATE, DELETE and MODIFY. 
Rule verification is largely based on set operations using the mini set library added with #34.
See commits and commit message for details, most notably b9ea81c0d6bb130ef3d77ff16c7bf046ee763257.

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by SantiagoTorres** [comment]
for some reason I think there's a creep from #34 even after I merged it. Could you rebase please?

File: in_toto/verifylib.go [hunk 1]
```diff
@@ -201,78 +163,131 @@ func VerifyArtifacts(items []interface{},
...
 
-		default: // Something's wrong
+		default: // Something wrong
 			return fmt.Errorf("VerifyArtifacts received an item of invalid type,"+
 				" elements of passed slice 'items' must be one of 'Step' or"+
 				" 'Inspection', got: '%s'", reflect.TypeOf(item))
 		}
+
+		// Use the item's name to extract the corresponding link
 		srcLinkMb, exists := itemsMetadata[itemName]
```
Comment (SantiagoTorres) [inline in_toto/verifylib.go]: Silly question, what does `Mb` stand for in this context? I keep on thinking of MetaBlock, but I don't think this is quite right?

File: in_toto/verifylib.go [hunk 2]
```diff
@@ -201,78 +163,131 @@ func VerifyArtifacts(items []interface{},
...
+			// on artifacts in that queue.  If a rule consumes an artifact (i.e. can
+			// be applied successfully), the artifact is removed from the queue. By
 			// applying a DISALLOW rule eventually, verification may return an error,
 			// if the rule matches any artifacts in the queue that should have been
 			// consumed earlier.
-			var queue []string
-			for name, _ := range artifacts {
-				queue = append(queue, name)
-			}
+			queue := verificationData["artifactPaths"].(Set)
```
Comment (SantiagoTorres) [inline in_toto/verifylib.go]: Curious, why do you need the parentheses here?

File: in_toto/verifylib.go (hunk 1 above)
Comment (lukpueh) [reply in_toto/verifylib.go]: Yep, Mb stands for metablock:
https://github.com/in-toto/in-toto-golang/blob/32abde0de57f3f002f55ac6319fe284cae384715/in_toto/verifylib.go#L64

File: in_toto/verifylib.go (hunk 2 above)
Comment (SantiagoTorres) [reply in_toto/verifylib.go]: Oh wow this syntax is horrible. Ok, thanks!

**Comment by lukpueh** [comment]
Thanks for the review, @SantiagoTorres!

**Pull request merged**